shoes-swt/lib/shoes/swt/arc_painter.rb

```diff
@@ -26,7 +26,33 @@ def draw(graphics_context)
                                     @obj.translate_top + @obj.element_top + sw / 2,
                                     @obj.element_width - sw,
                                     @obj.element_height - sw,
-                                    @obj.angle1, @obj.angle2 * -1)
+                                    start_angle, sweep)
+        end
+      end
+
+      # Shoes angles to run clockwise, with 0 at the 3 o'clock position. SWT
+      # instead runs counter-clockwise from 3 o'clock, hence we negate the
+      # start angle to translate from Shoes -> SWT
+      def start_angle
+        -@obj.angle1
+      end
+
+      # Shoes represents via a start and end angle, with the fill running
+      # clockwise from start to end. If start is greater than end, this just
+      # means that we wrap around the 3 o'clock-0 radians point as we run
+      # clockwise.
+      #
+      # SWT instead is start angle plus a sweep, and runs counter-clockwise.
+      # Hence we end up calculating a negative (clockwise) sweep. There's a
+      # small case difference depending on if we're crossing over the 0 point.
+      #
+      # See the specs for lots of good examples, with pictures, of this logic.
+      #
+      def sweep
+        if @obj.angle1 <= @obj.angle2
+          @obj.angle1 - @obj.angle2
+        else
+          @obj.angle1 - @obj.angle2 - 360
```

Comment:
maybe put 360 in a constant? It's sort of clear from the context but I still like to be more explicit about it :)